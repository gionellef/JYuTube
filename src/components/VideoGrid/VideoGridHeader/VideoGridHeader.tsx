import * as React from 'react';
import './VideoGridHeader.scss';

interface VideoGridHeaderProps {
  title?: string;
};

const VideoGridHeader : React.FC<VideoGridHeaderProps> = (props) => {
  return (
    <div className='video-grid-header'>
      <span className='title'>{ props.title }</span>
    </div>
  );
}

export default VideoGridHeader;