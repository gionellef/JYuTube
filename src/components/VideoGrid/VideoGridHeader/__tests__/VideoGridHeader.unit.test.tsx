import * as React from 'react';
import { shallow } from 'enzyme';

import VideoGridHeader from '../VideoGridHeader';

describe('VideoGridHeader', () => {

  test('renders VideoGridHeader without title', () => {
    const wrapper=shallow(<VideoGridHeader />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders VideoGridHeader with empty title', () => {
    const wrapper = shallow(<VideoGridHeader title='' />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders VideoGridHeader with title', () => {
    const wrapper = shallow(<VideoGridHeader title='Trending' />);
    expect(wrapper).toMatchSnapshot();
  });
});