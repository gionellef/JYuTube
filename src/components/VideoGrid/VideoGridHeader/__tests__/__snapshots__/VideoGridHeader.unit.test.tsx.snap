// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGridHeader renders VideoGridHeader with empty title 1`] = `
<div
  className="video-grid-header"
>
  <span
    className="title"
  />
</div>
`;

exports[`VideoGridHeader renders VideoGridHeader with title 1`] = `
<div
  className="video-grid-header"
>
  <span
    className="title"
  >
    Trending
  </span>
</div>
`;

exports[`VideoGridHeader renders VideoGridHeader without title 1`] = `
<div
  className="video-grid-header"
>
  <span
    className="title"
  />
</div>
`;
